src/Popup.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+import React, { isValidElement, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import type { AnyReactElement, ReactChildren } from 'react-flow-types';
+
+import icBEM from './utils/icBEM';
+import prefixClass from './utils/prefixClass';
+
+import Icon from './Icon';
+import Overlay from './Overlay';
+import escapable from './mixins/escapable';
+import renderToLayer from './mixins/renderToLayer';
+
+import './styles/Popup.scss';
+
+export const COMPONENT_NAME = prefixClass('popup');
+const ROOT_BEM = icBEM(COMPONENT_NAME);
+export const BEM = {
+    root: ROOT_BEM,
+    container: ROOT_BEM.element('container'),
+    body: ROOT_BEM.element('body'),
+    message: ROOT_BEM.element('message'),
+    button: ROOT_BEM.element('button'),
+    buttonsGroup: ROOT_BEM.element('buttons-group')
+};
+
+export type Props = {
+    icon?: string | ReactChildren,
+    message?: ReactChildren,
+    buttons?: AnyReactElement[],
+
+    /* eslint-disable react/require-default-props */
+    className?: string,
+    children?: ReactChildren,
+    /* eslint-enable react/require-default-props */
+};
+
+/**
+ * Render popup's icon
+ *
+ * @param  {String|Element} icon
+ * @return {Element}
+ */
+function renderPopupIcon(icon) {
+    if (!icon) {
+        return null;
+    }
+
+    if (typeof icon === 'string') {
+        return (
+            <Icon
+                large
+                color="blue"
+                type={icon} />
+        );
+    }
+
+    return icon;
+}
+
+/**
+ * Render popup's message
+ *
+ * @param  {Element|String} message
+ * @return {Element}
+ */
+function renderPopupMessage(message) {
+    if (!message) {
+        return null;
+    }
+
+    return (
+        <div className={BEM.message.toString()}>
+            {message}
+        </div>
+    );
+}
+
+/**
+ * Render popup's buttons
+ *
+ * @param  {Array} buttons
+ * @return {Array}
+ */
+function renderPopupButtons(buttons) {
+    if (!buttons || buttons.length === 0) {
+        return null;
+    }
+
+    const popupButtons = buttons.map((button) => {
+        // Render as expanded button
+        if (button && isValidElement(button)) {
```

Comment:
My bad: this breaks Flow type check, because Flow is unable to infer the existence of `button` through the runtime code inside `isValidElement`. 

Let me fix this.